meson.build

```diff
@@ -0,0 +1,1945 @@
+# -*- mode: meson -*-
+
+project('systemd', 'c',
+        version : '233',
+        license : 'LGPLv2+',
+        default_options: [
+          'c_std=gnu99',
+          'prefix=/usr',
+          'sysconfdir=/etc',
+          'localstatedir=/var',
+          ],
+        meson_version : '>= 0.39.1',
+       )
+
+# We need the same data in three different formats, ugh!
+# Also, for hysterical reasons, we use different variable
+# names, sometimes. Not all variables are included in every
+# set. Ugh, ugh, ugh!
+conf = configuration_data()
+conf.set_quoted('PACKAGE_STRING',  meson.project_name() + ' ' + meson.project_version())
+conf.set_quoted('PACKAGE_VERSION', meson.project_version())
+
+substs = configuration_data()
+substs.set('PACKAGE_URL',          'https://www.freedesktop.org/wiki/Software/systemd')
+substs.set('PACKAGE_VERSION',      meson.project_version())
+
+m4_defines = []
+
+#####################################################################
+
+if get_option('split-usr')
+  conf.set('HAVE_SPLIT_USR', 1)
+  rootprefix = ''
```

Comment:
That doesn't seem to work as intended.
When building with -Dsplit-usr=true, the binaries are still installed at /usr, like /usr/lib/systemd/systemd or /usr/bin/systemctl.

I'm also not sure if we only having -Dsplit-usr is sufficient. We probably need explicit switches for rootprefix and rootlibdir as we do for autotools. Those control different things really.